sympy/ntheory/digits.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import print_function, division
+
+import math
+import re
+
+from collections import Counter
+
+
+def count_digits(n, base=10):
+    """
+    A frequency counter for the digits of an integer ``n`` given in a base
+    ``base``.
+    """
+    base_func = {2: bin, 8: oct, 10: str, 16: hex}
+
+    if base in [2, 8, 10, 16]:
+        s = re.sub(r'0[a-z]{1}', '', base_func[base](n))
+        count = Counter(s)
+        return {int(k, base=16): v for k, v in count.items()}
+
+    m = math.floor(math.log(n, base)) + 1
+    count = dict()
+    for k in range(m):
+        d = (n // base ** k) % base
+        if d in count:
+            count[d] += 1
+        else:
+            count[d] = 1
+
+    return count
+
+
+def is_palindromic_integer(n, base=10):
+    """
+    Checks whether a given positive integer ``n``, specified either as a
+    numeric literal (binary, octal, decimal or hexadecimal) or a string, has
+    a symmetrical digit sequence, i.e. the sequence of its digits is the same
+    left to right vs right to left.
+    ::
+        1, 10 -> True
+        33, 10 -> True
+        919, 10 - > True
+        0b101, 2 -> True
+        0b01010010,2 -> False
+        0o7610167, 8 -> True
+        0o7611167, 8 -> False
+        0xFA0100AF, 16, False
+        0xFA0110AF, 16, True
+    """
```

Comment:
IMO, examples should be added here too.

Reply:
Yes. And it would be better to denote the examples as doctest format
For example, you can write the examples as
```
>>> is_palindromic_integer(919, 10)
True
```
And this is better than giving examples in pseudocodes in terms of testing, interactivity, and so on.

Reply:
THere are a few typos in the docstrings, which I will correct and edit in a final commit, once this build passes. Any final commits will be squashed also.